File: AlgoAgentXAPI/app/schemas/profile_settings.py
from datetime import datetime
from typing import Any
from uuid import UUID
from pydantic import BaseModel, Field

class ProfileStats(BaseModel):
    total_backtests: int = 0
    connected_brokers: int = 0
    active_subscription: str | None = None
    credit_balance: int = 0
    admin_console_access: bool = False

class ProfileResponse(BaseModel):
    id: UUID
    email: str
    role: str
    full_name: str | None = None
    fullname: str | None = None
    mobile: str | None = None
    company: str | None = None
    created_at: datetime | None = None
    last_login_at: datetime | None = None
    account_status: str = "active"
    stats: ProfileStats = Field(default_factory=ProfileStats)

class ProfileUpdate(BaseModel):
    full_name: str | None = Field(default=None, max_length=150)
    fullname: str | None = Field(default=None, max_length=150)
    mobile: str | None = Field(default=None, max_length=30)
    company: str | None = Field(default=None, max_length=150)

class ChangePasswordRequest(BaseModel):
    current_password: str = Field(..., min_length=1)
    new_password: str = Field(..., min_length=8)

class SettingsResponse(BaseModel):
    preferences: dict[str, Any] = Field(default_factory=dict)
    notifications: dict[str, Any] = Field(default_factory=dict)
    safety: dict[str, Any] = Field(default_factory=dict)
    admin_alerts: dict[str, Any] = Field(default_factory=dict)

class SettingsUpdate(BaseModel):
    preferences: dict[str, Any] | None = None
    notifications: dict[str, Any] | None = None
    safety: dict[str, Any] | None = None
    admin_alerts: dict[str, Any] | None = None

File: AlgoAgentXApp/app/lib/api/profile-settings.ts
import axiosInstance from "../axios";

export type ProfileStats = {
  total_backtests: number;
  connected_brokers: number;
  active_subscription?: string | null;
  credit_balance: number;
  admin_console_access?: boolean;
};

export type Profile = {
  id: string;
  email: string;
  role: string;
  full_name?: string | null;
  fullname?: string | null;
  mobile?: string | null;
  company?: string | null;
  created_at?: string | null;
  last_login_at?: string | null;
  account_status?: string | null;
  stats?: ProfileStats;
};

export type AccountSettings = {
  preferences: Record<string, any>;
  notifications: Record<string, any>;
  safety: Record<string, boolean>;
  admin_alerts: Record<string, any>;
};

export const profileSettingsApi = {
  getProfile: async (): Promise<Profile> => {
    const response = await axiosInstance.get("/api/v1/profile/me");
    return response.data;
  },
  updateProfile: async (body: { full_name?: string; mobile?: string; company?: string }): Promise<Profile> => {
    const response = await axiosInstance.patch("/api/v1/profile/me", body);
    return response.data;
  },
  changePassword: async (body: { current_password: string; new_password: string }): Promise<{ message: string }> => {
    const response = await axiosInstance.post("/api/v1/profile/change-password", body);
    return response.data;
  },
  getSettings: async (): Promise<AccountSettings> => {
    const response = await axiosInstance.get("/api/v1/settings/me");
    return response.data;
  },
  updateSettings: async (body: Partial<AccountSettings>): Promise<AccountSettings> => {
    const response = await axiosInstance.patch("/api/v1/settings/me", body);
    return response.data;
  },
};
